build.gradle

```diff
@@ -1,114 +1,147 @@
 /// Gradle Build script for 18xx Rails program (rails.sf.net)
 
-import org.apache.tools.ant.filters.*
 import java.text.SimpleDateFormat
 
-
-
-// Addition of JavaProfile plugin to read Java property files
-// reference: https://github.com/unsaved/gradle-javaPropFile-plugin
-
-buildscript {
-    repositories { jcenter() }  // Before plugin v. 1.0.0 use mavenCentral()
-    dependencies {
-        classpath 'com.admc:gradle-javaPropFile-plugin:1.0.0'
-        // For strictly repeatable builds, replace 'latest.integration'
-        // with the latest literal GA version label.
-    }
-}
-
-
-// Addition of launch4j gradle plugin
-// refer to https://github.com/TheBoegl/gradle-launch4j
-
-buildscript {
-  repositories {
-    maven {
-      url "https://plugins.gradle.org/m2/"
-    }
-  }
-  dependencies {
-    classpath "gradle.plugin.edu.sc.seis.gradle:launch4j:2.3.0"
-  }
-}
-
-// Addition of shadow plugin to create one jars
-// reference: https://github.com/johnrengelman/shadow
-plugins {
-  id 'java' // or 'groovy' Must be explicitly applied
-  id 'com.github.johnrengelman.shadow' version '5.0.0'
-}
-
-// Addition of macAppBundle
-// https://github.com/crotwell/gradle-macappbundle
 plugins {
+  id 'java'
+  id 'maven-publish'
+  // https://github.com/johnrengelman/shadow
+  id 'com.github.johnrengelman.shadow' version '5.2.0'
```

Comment:
This is for the creation of fat-jars right?
I'm currently not sure whether we should take this direction.
`jpackage` seems to be able to work with a collection of multiple jars, i.e. a `lib` folder where all the dependencies are located

Reply:
Hmmm.. had replied to this but guess I didn't hit submit.

I'm not very familiar with `jpackage` so bear with me and please correct any misunderstandings. But based on your comment it sounds like it's able to bundle up all dependencies but stores them in a directory tree vs generating a single output file? If that is correct then I'd suggest we still need to be able to generate a fat jar. We can't assume that someone, especially someone technical in nature (which I suspect happens to be a large chunk of the 18xx player population!) is going to want to use an app or rpm. And the overhead and supporting that additional generation is minor in nature.

Reply:
> We can't assume that someone, especially someone technical in nature (which I suspect happens to be a large chunk of the 18xx player population!) is going to want to use an app or rpm.

He does not need to, he could also load the external dependencies using a `-classpath`/`-cp` parameter when calling `java -jar rails.jar`. Ideally we would wrap this in a shell-/bash-script.

Still this is only an idea. We could also go with a far-jar, I'm not sure how both approaches compare against each other.